src/lib.rs

```diff
@@ -382,11 +382,6 @@ impl<T: Translation> Mapping<T> {
             self.check_range_bbm(range, &c)?;
         }
         self.root.map_range(range, pa, flags, constraints)?;
-        #[cfg(target_arch = "aarch64")]
-        // SAFETY: This is just a memory barrier.
-        unsafe {
-            asm!("dsb ishst");
```

Comment:
Does the compiler generate a DSB for each atomic write, or is this unnecessary for another reason?

Reply:
No, and there is no need - the acquire/release semantics of AtomicUsize will ensure that special load and store instructions are used that have the appropriate barrier behavior.

E.g., [LDAR](https://developer.arm.com/documentation/ddi0596/2020-12/Base-Instructions/LDAR--Load-Acquire-Register-) will be used to implement `AtomicUsize::load(..., Ordering::Acquire)`

Reply:
Note that `Descriptor::set()` uses `Ordering::Relaxed`, but `Mapping::activate()` does have release semantics, so when building a set of inactive page tables and activating them, all new contents are guaranteed to be observable. But come to think of it, that does mean `mark_active()` should be called before updating TTBR. I'l fix that.

When updating live page tables, the relaxed semantics for setting entries is fine - BBM rules guarantee that observing those stores in a different order does not result in inconsistencies.